components/query/query-vector.service.js

```diff
@@ -82,6 +81,38 @@ export default function (
     }
     me.createFeatureAttributeList();
   });
+  $rootScope.$on(
+    'vectorQuery.featureSelected',
+    HsUtilsService.debounce(
+      (event, feature) => {
+        const layer = feature.getLayer(HsMapService.map);
+        if (angular.isDefined(layer) && layer.get('onFeatureSelected')) {
+          const originalFeature = me
+            .getSingleFeature(feature)
+            .map((feature) => feature)
```

Comment:
What is this `map()` actually useful for?

Reply:
Map is needed to create a clone of feature, because other way I was experiencing that missing feature from the cluster source. Of course this would not be needed, if I wouldn't use .pop(). .pop() takes the object from memory reference and 'pulls it out', which caused that styling error. If You do think that it should be removed, I will do it!

Reply:
onFeatureSelected: function (feature) {
    if (feature.get('myDirectLink')) {
      window.open(feature.get('myDirectLink'), '_blank');
    }
  }, this is the code I used in examples/full/app.js for testing

Reply:
Ah, interesting. In that case, I would write it in a way I suggest in the new comment.
FTR, the .map() does not create a copy of `feature` either, it copies the reference to the same object, see e.g. https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/